BrickHack-Mobile/Controllers/ScheduleTableViewController.swift

```diff
@@ -270,4 +262,126 @@ class ScheduleTableViewController: UITableViewController {
         return .leastNormalMagnitude
     }
 
+    // Set on a Timer to run every so often
+    // Updates the timer dispay to highlight the current event.
+    // Note: The topmost event is ALWAYS highlighted, even if it has not occured yet.
+    // (This is done in the dequeueCell tableview delegtae method)
+    @objc func refreshTimeline() {
+
+        // First, get an updated verison of the schedule
+        // (Handles UI, threaded)
+        updateSchedule() {
+
+            // Don't parse if no sections available
+            // (Sanity guard, shouldn't happen)
+            guard ScheduleParser.sectionCount >= 1 else {
+                print("nope")
+                return
+            }
+
+            // Start at -1 as we look at the next index to see if the current is over yet
+            // (Updating each section as "current" as we go)
+            print("sectionCount: \(ScheduleParser.sectionCount)")
+            for sectionIndex in 0..<ScheduleParser.sectionCount {
+
+                // Grab the next section's date
+                let events = self.timelineEvents.filter({ $0.event.section == sectionIndex })
+                let sectionDate = events.first!.event.time
+
+                // If greater, STOP. We are at the current section.
+                if (sectionDate > Date(timeIntervalSinceNow: 0)) {
+                    break
+                }
+
+                // Otherwise, go on to configure this current section as "passed"
+                self.timelineEvents.filter({ $0.event.section == sectionIndex }).forEach { timelineEvent in
+
+                    print("updated \(timelineEvent)")
+                    timelineEvent.allColor = self.backColor
+                }
+
+                // @TODO: Set the current event with a TimelinePoint
+                let mostCurrentEvent = self.timelineEvents.filter({ $0.event.section == sectionIndex }).last
+                mostCurrentEvent?.timelinePoint = TimelinePoint(color: self.backColor, filled: true)
+
+                // Remove it from the previous section
+                let lastCurrentEvent = self.timelineEvents.filter({ $0.event.section == sectionIndex - 1}).last
+                lastCurrentEvent?.timelinePoint = nil
+            }
+
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+
+        }
+
+    }
+
+    // Updates the listing of events from the Google Sheet
+    func updateSchedule(completion: @escaping () -> ()) {
```

Comment:
These escaping closures invite horrible memory management, and there can be a bit of a stutter when the schedule updates. Minor issue given that in prod it'll update once each 30 minutes or so, which won't be that noticeable.